tests_metricflow/snapshots/test_custom_granularity.py/SqlQueryPlan/DuckDB/test_offset_metric_with_custom_granularity__plan0_optimized.sql

```diff
@@ -32,4 +29,23 @@ FROM (
     subq_12.ds = subq_13.ds
   GROUP BY
     subq_13.martian_day
-) subq_17
+)
+
+, cm_5_cte AS (
+  -- Compute Metrics via Expressions
+  SELECT
+    booking__ds__martian_day
+    , bookings_5_days_ago AS bookings_5_day_lag
+  FROM (
+    -- Read From CTE For node_id=cm_4
+    SELECT
+      booking__ds__martian_day
+      , bookings_5_days_ago
+    FROM cm_4_cte cm_4_cte
+  ) subq_17
+)
+
+SELECT
```

Comment:
Do we need to rename these columns at the end? We already did that step in cm_5

Reply:
This is the result of following the rule, if multiple metrics are computed in a query (parts of a derived query are each considered a metric computation), have each metric computation is out into a CTE. We can change the rule to get different behavior.